bentoml/cli/azure_function.py

```diff
@@ -0,0 +1,423 @@
+DEFAULT_MAX_BURST = 20
+DEFAULT_PREMIUM_PLAN_SKU = 'EP1'
+DEFAULT_FUNCTION_AUTH_LEVEL = 'function'
+
+
+def get_azure_function_sub_command():
+    # pylint: disable=unused-variable
+
+    @click.group(
+        name='azure-function',
+        help='Commands for Azure function BentoService deployment',
+        cls=BentoMLCommandGroup,
+    )
+    def azure_function():
+        pass
+
+    @azure_function.command(help='Deploy BentoService to Azure function')
+    @click.option(
+        '-n',
+        '--namespace',
+        type=click.STRING,
+        help='Deployment namespace managed by BentoML, default value is "dev" which'
+        'can be changed in BentoML configuration yatai_service/default_namespace',
+    )
+    @click.argument('name', type=click.STRING)
+    @click.option(
+        '-b',
+        '--bento',
+        '--bento-service-bundle',
+        type=click.STRING,
+        required=True,
+        callback=parse_bento_tag_callback,
+        help='Target BentoService to be deployed, referenced by its name and version '
+        'in the format of name:version. For example: "iris_classifier:v1.2.0"',
+    )
+    @click.option(
+        '--location',
+        type=click.STRING,
+        help='The Azure location name for the deployment',
+        required=True,
+    )
+    @click.option(
+        '--min-instances',
+        type=click.INT,
+        default=DEFAULT_MIN_INSTANCE_COUNT,
+        help='The minimum number of workers for the deployment. The default value is 1',
+    )
+    @click.option(
+        '--max-burst',
+        type=click.INT,
+        default=DEFAULT_MAX_BURST,
+        help='The maximum number of elastic workers for the deployment. '
+        'The default value is 3',
+    )
+    @click.option(
+        '--premium-plan-sku',
+        type=click.STRING,
+        default=DEFAULT_PREMIUM_PLAN_SKU,
+        help='The Azure premium SKU for the deployment. Options are EP1, EP2, and EP3.'
+        ' The default value is EP1',
+    )
+    @click.option(
+        '-l',
+        '--labels',
+        type=click.STRING,
+        callback=parse_labels_callback,
+        help='Key:value pairs that are attached to deployments and intended to be used '
+        'to specify identifying attributes of the deployments that are meaningful to '
+        'users. Multiple labels are separated with `,`',
+    )
+    @click.option(
+        '--function-auth-level',
+        type=click.STRING,
+        default=DEFAULT_FUNCTION_AUTH_LEVEL,
+        help='The authorization level for the deployed Azure function. '
+        'Options are "anonymous", "function", and "admin". Default value is '
+        'function',
+    )
+    @click.option('-o', '--output', type=click.Choice(['json', 'yaml']), default='json')
+    @click.option(
+        '--wait/--no-wait',
+        default=True,
+        help='Wait for apply action to complete or encounter an error.'
+        'If set to no-wait, CLI will return immediately. The default value is wait',
```

Comment:
", CLI will exit without waiting until it has verified cloud resource allocation"?